src/Microsoft.Extensions.DependencyInjection.Abstractions/IDesignTimeServiceProviderFactory.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// A factory for creating the application's <see cref="IServiceProvider"/> at design time. Implement this interface
+    /// to enable design-time services that need access to the application's services. Design-time services will
+    /// automatically discover implementations of this interface.
+    /// </summary>
+    public interface IDesignTimeServiceProviderFactory
```

Comment:
@Eilon don't think it is needed at runtime. For as far as I've thought it through so far, the discovery would be done by XDesProc only if you open a XAML file, to instantiate views and their dependencies at design time without any bootstrapping having done already, allowing constructor arguments to be resolved. There is also overhead in finding it, which is especially important for apps and not strictly necessary since there is code that definitely runs before navigation.